code/game/objects/items/_inventory.dm

```diff
@@ -63,12 +63,12 @@
 
 /obj/item/proc/is_worn()
 	if (ismob(loc))
-		return !(equip_slot in list(slot_none, slot_l_hand, slot_r_hand))
+		return !(equip_slot in list(slot_none, slot_l_hand, slot_r_hand,slot_robot_equip_1,slot_robot_equip_2,slot_robot_equip_3))
```

Comment:
You added a new `unworn_slots` global but the proc called `is_worn` does not take it into account, that does not follow.